rescheduler/rescheduler.go

```diff
@@ -112,6 +123,11 @@ func main() {
 				if len(criticalPods) > 0 {
 					for _, pod := range criticalPods {
 						glog.Infof("Critical pod %s is unschedulable. Trying to find a spot for it.", podId(pod))
+						k8sApp := "unknown"
+						if l, found := pod.ObjectMeta.Labels["k8s-app"]; found {
```

Comment:
I totally agree that it makes sense. You can try to open an issue in the main repo and cc Cluster Lifecycle team and Brian Grant. But let's start with something. In the worsts case it will be just "unknown".